rmcreyes/cpen391-hardware-app
# Setup Hardware App

# :ship: Pull Request

## :question: Purpose

Set up the layout and navigation for view in the hardware app
Add basic Bluetooth functionality and structure to read/write data. Defined the TODO methods for Bluetooth implementation
- NOTE: Currently will try to connect to the first device in the paired list
- NOTE: Will not do anything if there are no paired devices

<p float="left">
<img src="https://user-images.githubusercontent.com/42983386/111249367-d6491800-85c8-11eb-8ddf-3ec7f2974f60.png" width="130" height="300">
<img src="https://user-images.githubusercontent.com/42983386/111249364-d5b08180-85c8-11eb-991c-a018866fea20.png" width="130" height="300">
<img src="https://user-images.githubusercontent.com/42983386/111249371-d77a4500-85c8-11eb-8f31-b7a21354cc13.png" width="130" height="300">
<img src="https://user-images.githubusercontent.com/42983386/111249370-d6e1ae80-85c8-11eb-9155-1d9bac2042ec.png" width="130" height="300">
<img src="https://user-images.githubusercontent.com/42983386/111249368-d6e1ae80-85c8-11eb-9626-4df33cd8718f.png" width="130" height="300">
</p>


aired devices


## :hammer: Validation Strategy
Manual testing for basic flow between views
- Initialize the app, click on the `enter plate` button
   - Ensure that you are navigated to the plate fragment
- Enter a 6-digit string and click` confirm`
- Ensure that you are navigated to the `Confirm` fragment
   - Ensure that if you click on the `modify` button, you are navigated back to re-enter your plate number
   - Ensure that if you click on the `confirm` button, you are navigated to the `payment` fragment
- Ensure that if you click `confirm`, you are navigated to the `occupied` fragment
     - Ensure that the meter says "Occupied" along with the entered plate number. 

## :tickets: Ticket(s)

Closes #5
